### PR TITLE
Feature/170178161/simplify dashboard widgets hiro ui

### PR DESCRIPTION
**Pull request by prapakou**
#170178161

- replace switch-case clause that imports specific widget component with a conditional one
- all widget components are made with default export
- common stuff is replaced to dashboard/shared

### CODE REVIEW

File: packages/bootstrap/src/components/dashboard/widgets/index.ts [hunk 1]
```diff
@@ -22,75 +22,39 @@ import { Widgets } from '../types';
...
+
+    return lazy(() =>
+      importCurrentWidget()
+        .then((module) => ({
+          default: module.default,
+        }))
+        .catch(() => {
+          return flexBoxImport();
+        }),
+    );
```
Comment (EKazachenko) [inline packages/bootstrap/src/components/dashboard/widgets/index.ts]:
```
lazy(importCurrentWidget.then(...
```

File: packages/bootstrap/src/components/dashboard/shared/Welcome/WelcomePresenter.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { Box } from '../../../elements';
 import { SessionsButton } from './SessionsButton';
 import { SessionsButtonText } from './SessionsButtonText';
-import { WidgetTitle } from '../../shared';
+import { WidgetTitle } from '../index';
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/shared/Welcome/WelcomePresenter.tsx]: image

File: packages/bootstrap/src/components/dashboard/widgets/index.ts [hunk 3]
```diff
@@ -22,75 +22,39 @@ import { Widgets } from '../types';
...
+  StatisticTeaching: () => import('./StatisticTeaching'),
+  StatisticConversion: () => import('./StatisticConversion'),
+  ConversionSessions: () => import('./ConversionSessions'),
+  TeachingSessions: () => import('./TeachingSessions'),
+  WelcomeTeaching: () => import('./WelcomeTeaching'),
+  WelcomeConversion: () => import('./WelcomeConversion'),
+  TopTeachingContributors: () => import('./TopTeachingContributors'),
+  TopConverters: () => import('./TopConverters'),
+  YourKis: () => import('./YourKis'),
+  Demo: () => import('./Demo'),
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/index.ts]: I think we should delete the `Demo` widget.

File: packages/bootstrap/src/components/dashboard/widgets/index.ts [hunk 4]
```diff
@@ -22,75 +22,39 @@ import { Widgets } from '../types';
...
 };
 
+const flexBoxImport: WebpackModule = () => import('../../elements/Flex/Box');
+
+export const useWidget = (type?: Widgets) =>
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/index.ts]: `type` should be required

File: packages/bootstrap/src/components/dashboard/widgets/index.ts [hunk 5]
```diff
@@ -22,75 +22,39 @@ import { Widgets } from '../types';
...
+  TeachingSessions: () => import('./TeachingSessions'),
+  WelcomeTeaching: () => import('./WelcomeTeaching'),
+  WelcomeConversion: () => import('./WelcomeConversion'),
+  TopTeachingContributors: () => import('./TopTeachingContributors'),
+  TopConverters: () => import('./TopConverters'),
+  YourKis: () => import('./YourKis'),
+  Demo: () => import('./Demo'),
 };
 
+const flexBoxImport: WebpackModule = () => import('../../elements/Flex/Box');
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/index.ts]: Delete this fallback

File: packages/bootstrap/src/components/dashboard/widgets/index.ts [hunk 6]
```diff
@@ -22,75 +22,39 @@ import { Widgets } from '../types';
...
+      type && widgetImportMap[type] ? widgetImportMap[type] : flexBoxImport;
+
+    return lazy(() =>
+      importCurrentWidget()
+        .then((module) => ({
+          default: module.default,
+        }))
+        .catch(() => {
+          return flexBoxImport();
+        }),
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/index.ts]: `return lazy(widgetImportMap[type]);`

File: packages/bootstrap/src/components/dashboard/widgets/index.ts [hunk 7]
```diff
@@ -22,75 +22,39 @@ import { Widgets } from '../types';
...
-        );
-      case 'Demo':
-        return lazy(() =>
-          import('./Demo').then((module) => ({ default: module.Demo })),
-        );
-      default:
-        if (type) {
-          console.warn(`Async component '${type}' not supported`);
-        }
+type WebpackModule = () => Promise<{ default: React.ComponentType }>;
```
Comment (miraage) [inline packages/bootstrap/src/components/dashboard/widgets/index.ts]: This module has nothing to do with webpack.
It's more like `AsyncComponentFactory`

**Comment by alex-melnik** [comment]
Could you move `SessionList` to `shared`? It looks like presenter

File: packages/bootstrap/src/components/dashboard/widgets/index.ts (hunk 3 above)
Comment (alex-melnik) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: @yury-klika ^^

Comment (yury-klika) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: I'm agree. As soon as we dropped Storybook support of such complex components as Widgets - the Demo widget is no more required

File: packages/bootstrap/src/components/dashboard/shared/Welcome/WelcomePresenter.tsx (hunk 2 above)
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/shared/Welcome/WelcomePresenter.tsx]: fixed

File: packages/bootstrap/src/components/dashboard/widgets/index.ts (hunk 3 above)
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: fixed

File: packages/bootstrap/src/components/dashboard/widgets/index.ts (hunk 4 above)
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: fixed

File: packages/bootstrap/src/components/dashboard/widgets/index.ts (hunk 5 above)
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: fixed

File: packages/bootstrap/src/components/dashboard/widgets/index.ts (hunk 6 above)
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: fixed

File: packages/bootstrap/src/components/dashboard/widgets/index.ts (hunk 7 above)
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/widgets/index.ts]: fixed

**Comment by prapakou** [comment]
> Could you move `SessionList` to `shared`? It looks like presenter

fixed

File: packages/bootstrap/src/components/dashboard/widgets/ConversionSessions/index.ts [hunk 8]
```diff
@@ -1 +1,3 @@
-export * from './ConversionSessions';
+import { ConversionSessions } from './ConversionSessions';
+
+export default ConversionSessions;
```
[the remark this answers is not in the document]
Comment (prapakou) [reply packages/bootstrap/src/components/dashboard/widgets/ConversionSessions/index.ts]: @yury-klika we decided to stay it as it is. so we have index.ts with module exports in every widget folder.

**Pull request merged**